### PR TITLE
Add some types

### PR DESCRIPTION
**Pull request by flaviut**
While I was reading over this, I had some trouble understanding the types involved, so I decided to write them in myself.

I've never worked with any of the typecheckers, but these types should at least be helpful for understanding and for IDE autocomplete

### CODE REVIEW

File: sigrokdecode/__init__.py [hunk 1]
```diff
@@ -22,7 +23,23 @@ class OutputType(Enum):
 OUTPUT_LOGIC = OutputType.SRD_OUTPUT_LOGIC
 OUTPUT_META = OutputType.SRD_OUTPUT_META
 
-SRD_CONF_SAMPLERATE = 1
+class MetadataKeys(Enum):
```
Comment (flaviut) [inline sigrokdecode/__init__.py:26]: Breaking change

File: sigrokdecode/srzip.py [hunk 2]
```diff
@@ -30,9 +32,9 @@
 class SrZipInput(Input):
     name = "srzip"
     desc = "srzip session file format data"
-    def __init__(self, filename, initial_state=None):
+    def __init__(self, file: Union[str, PathLike[str], IO[bytes]], initial_state=None):
```
Comment (flaviut) [inline sigrokdecode/srzip.py:35]: breaking change renaming filename -> file

File: sigrokdecode/srzip.py [hunk 3]
```diff
@@ -190,12 +192,12 @@ def get_analog_values(self, samplenum):
 class SrZipOutput(Output):
     name = "srzip"
     desc = "srzip session file format data"
-    def __init__(self, filename, driver, logic_channels=[], analog_channels=[], decoders=[]):
+    def __init__(self, file: Union[str, PathLike[str], IO[bytes]], driver, logic_channels=[], analog_channels=[], decoders=[]):
```
Comment (flaviut) [inline sigrokdecode/srzip.py:195]: breaking change renaming filename -> file

**Review by tannewt — APPROVED** [review]
Looks great! Thanks for the comments too!

**Pull request merged**